<h1>
    Handling catchable exceptions in user processes
</h1>

<p>
    The message broker "wraps" all user code in a try-catch to catch exceptions.
    <br />The error and stack trace will be persisted onto the message.
</p>

<h2>
    Procedure
</h2>

<ol>
    <li>Create a new apex class:
<pre>
global with sharing class Error extends BaseProcess {
    
    global class ESB {
        String Description = 'This process throws a catchable exception.';
    }
    
    override protected List&lt;Map&lt;String,Object&gt;&gt; execute(Map&lt;String,Object&gt; inputEvent) {
        Integer undefined = 0 / 0;
    }
    
}
</pre>
    </li>
    
    <li>Create a new Sequence 'ErrorTest' containing a 'Error' step in position 1.</li>
    <li>Click <strong>Enqueue Test Message</strong> to process a message, which should fail.</li>
</ol>

<h2>
    Outcomes
</h2>

<ol>
    <li>The monitor should graph a single 'Exception' message for 'Error'.</li>
    <li>The messages listview 'Exception' should list one message.</li>
    <li>The message itself should have a 'Cause' stack trace and an 'Exception' of 'Divide by zero...'</li>
</ol>
